add simplified dots notebook

sergpolly opened the pull request.

gfudenberg commented on dots.ipynb:
```diff
@@ -0,0 +1,637 @@
+{
```
<p>Welcome to the dot calling notebook! </p><p> </p><p>Punctate pairwise peaks of enriched contact frequency are a prevalent feature of mammalian interphase contact maps. These features are also referred to as ‘dots’ or ‘loops’ in the literature, and can appear either in isolation or as parts of grids and at the corners of domains. </p><p> </p><p>HiCCUPS, proposed in <a href="https://doi.org/10.1016/j.cell.2014.11.021" target="_blank">Rao et al. 2014</a>, is a common approach for detecting these features in Hi-C maps. However, HICCUPS is written in Java, making it hard to modify the parameters for specific steps. </p><p> </p><p>Cooltools implements a similar approach for calling dots  in Python. This makes it easy to vary parameters and processing steps for Hi-C or Micro-C datasets.</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/22/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

gfudenberg replied:
<p>To call dots, we need an input cooler file with Hi-C data, and regions for calculation of expected (e.g. chromosomes or chromosome arms).</p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/22/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

gfudenberg replied:
<p>We first call dots with default parameters (i.e. similar to HiCCUPs). Later we then explore the various parameters than can be easily modified in cooltools.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/22/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

gfudenberg replied:
<p>To visualize the results of this dot calling, we overlay small rectangles at the positions of the called dots over the HiC map. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/22/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='9'/>

gfudenberg replied:
<p>Dot-calling returns pixels that are enriched relative to some local background. Such pixels often come in groups ("clusters"). By default <code>dots()</code> picks a single representative for each cluster (i.e. centroid). However, cooltools users can easily turn clustering off for debugging or alternative clustering approaches:</p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/22/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='12'/>

gfudenberg replied:
<p>The visualization below compares </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/22/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='14'/>

gfudenberg commented on dots.ipynb:
```diff
@@ -0,0 +1,637 @@
+{
```
<p>should this be bold? or moved into the next cell as a #comment? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/22/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

sergpolly replied:
<p>moved it</p>

gfudenberg commented:
<p>This looks like a great start!</p><p>I'd also add some overview of the approach where dots() is first called, e.g. along the lines of the docstring https://github.com/open2c/cooltools/blob/45a56dfd07b8058b908763cc7c8e7db3fc52875e/cooltools/api/dotfinder.py#L26 (though perhaps streamlined a bit)</p>

sergpolly commented:
<p><a href="https://github.com/gfudenberg" target="_blank">gfudenberg</a> I added that description in the beginning of the notebook - would appreciate if you could help me iron out the text</p><p>I've also changed the description near the discussion about kernels - could you please check that one out as well ?</p><p> </p><p>do you think we could merge it once that's done ? I'll figure out CTCF-related stuff after that</p>

gfudenberg commented:
looks great! merged!
